src/Hazelcast.Net/HazelcastClient.Objects.cs

```diff
@@ -28,6 +30,9 @@ internal partial class HazelcastClient // Distributed Objects
         private readonly ISequence<long> _lockReferenceIdSequence = new Int64Sequence();
         private ICPSubsystem _cpSubsystem;
 
+        // FIXME [Oleksii] discuss if lazy is needed
```

Comment:
Want to check if lazy initialization is needed here. `SqlService` is a simple relatively lightweight class, so probaly not. On the other hand `CPSubsystem` is similar but uses (thread-unsafe) lazy initialization in it's accessor.

Reply:
I think lazy initialization is good just to try to create as few objects as possible when we don't need them. And yes, the `CPSubsystem` initialization is not thread-safe because the class is stateless so it's not that important. But maybe it's cleaner to make it thread-safe with a lock, or a `Lazy<T>`, and do the same for `SqlService`?

Reply:
Well, `Lazy<T>` is an object 😉. And straightforward initialization approach currently used by `CPSubsystem` may lead to several instances be created. By the way, `SqlService` is also stateless, at least the same way as `CPSubsystem` is.

Reply:
Agreed on creating them immediately.